src/kinetics/LinearBurkeRate.cpp

```diff
@@ -0,0 +1,354 @@
+        rateObj_M = PlogRate(colliders[0], rate_units);
+        dataObj_M = PlogData();
+    }
+    else if (colliders[0].hasKey("Troe")) {
+        // Value of "type" is unimportant; just needed to make falloff.cpp run
+        colliders[0]["type"];
+        rateObj_M = TroeRate(colliders[0], rate_units);
+        dataObj_M = FalloffData();
+    }
+    else if (colliders[0].hasKey("pressure-range")) {
+        rateObj_M = ChebyshevRate(colliders[0], rate_units);
+        dataObj_M = ChebyshevData();
+    }
+    else {
+        throw InputFileError("LinearBurkeRate::setParameters", m_input,
+            "The M-collider must be specified in a pressure-dependent-Arrhenius (PLOG), falloff (Troe form), or Chebyshev format.");
+    }
+    string eig_eps_key;
+    if (colliders[0].hasKey("eig0") && !colliders[0].hasKey("eps")) {
+        eig_eps_key = "eig0";
+    }
+    else if (colliders[0].hasKey("eps") && !colliders[0].hasKey("eig0")) {
+        eig_eps_key = "eps";
+    }
+    else{
+        throw InputFileError("LinearBurkeRate::setParameters", m_input,
+            "Cannot have both eig0 and eps chosen for M. Any additional colliders must make same choice as that of M. Please review implementation guide.");
+    }
+    AnyMap params;
+    params["A"] = 1.0;
+    params["b"] = 0.0;
+    params["Ea"] = 0.0;
+    epsObj_M = ArrheniusRate(AnyValue(params), colliders[0].units(), eps_units);
+    // Start at 1 because index 0 is for "M"
+    for (size_t i = 1; i < colliders.size(); i++){
+        if (!colliders[i].hasKey("collider")) {
+                throw InputFileError("LinearBurkeRate::setParameters", m_input,
+                    "Incorrect YAML input for LMR-R reaction. Please review implementation guide.");
+        }
+        if (!colliders[i].hasKey(eig_eps_key)) {
+            throw InputFileError("LinearBurkeRate::setParameters", m_input,
+                "All collider strengths must be defined uniformly as either eig0 or eps. No mixing and matching is allowed.");
+        }
+        // Save data to colliderInfo, which will make it accessible by getParameters
+        colliderInfo.insert({colliders[i]["collider"].as<string>(), colliders[i]});
+        colliderNames.push_back(colliders[i]["collider"].as<string>());
+        ArrheniusRate epsObj_i;
+        // eig0 and eps are ONLY interchangeable due to the requirement that eps_M have parameters {A: 1, b: 0, Ea: 0}
+        params["A"] = colliders[i][eig_eps_key]["A"].asDouble() / colliders[0][eig_eps_key]["A"].asDouble();
+        params["b"] = colliders[i][eig_eps_key]["b"].asDouble() - colliders[0][eig_eps_key]["b"].asDouble();
+        params["Ea"] = colliders[i][eig_eps_key]["Ea"].asDouble() - colliders[0][eig_eps_key]["Ea"].asDouble();
+        epsObj_i = ArrheniusRate(AnyValue(params), colliders[i].units(), eps_units);
+        if (colliders[i].hasKey("rate-constants")) {
+            rateObjs.push_back(PlogRate(colliders[i], rate_units));
```

Comment:
Rather than checking which fields the collider map has to decide what kind of rate it is, I think you should rely on the same method as is used for the base rate types, where the `type` field is examined (and in this case should be `pressure-dependent-Arrhenius`)

Reply:
By using duck typing, a k(T,P) parameterization can be provided for individual colliders (besides just M) without needing to specify a "type" for each one, which may get messy. This code permits mixing of different k(T,P) parameterizations even within the same reaction, without requiring the user to additionally specify "type: falloff", "type: Chebyshev", etc for every collider instance